src/Cdn/Cdn.Test/ScenarioTests/CustomDomainTests.ps1

```diff
@@ -152,5 +152,56 @@ function Test-CustomDomainGetRemoveWithStoppedEndpoint
 
     Assert-ThrowsContains { Get-AzCdnCustomDomain -CustomDomainName $customDomainName -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName } "NotFound"
 
-	Remove-AzResourceGroup -Name $resourceGroup.ResourceGroupName -Force
+    Remove-AzureRmResourceGroup -Name $resourceGroup.ResourceGroupName -Force
+}
+
+<#
+.SYNOPSIS
+Enable Https for custom domain with running endpoint
+#>
+function Test-CustomDomainEnableHttpsWithRunningEndpoint
+{
+  # Hard-coding host and endpoint names due to requirement for DNS CNAME
+    $endpointName = "SslTestAkamaiEP"
+    $hostName = "ssltestakamai.dustydog.us"
+
+    $customDomainName = getAssetName
+
+    $profileName = getAssetName
+    $resourceGroup = TestSetup-CreateResourceGroup
+    $resourceLocation = "EastUS"
+    $profileSku = "Standard_Akamai"
+    $tags = @{"tag1" = "value1"; "tag2" = "value2"}
+    $createdProfile = New-AzCdnProfile -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName -Location $resourceLocation -Sku $profileSku -Tag $tags
+
+    $originName = getAssetName
+    $originHostName = "www.microsoft.com"
+    $createdEndpoint = New-AzCdnEndpoint -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName -Location $resourceLocation -OriginName $originName -OriginHostName $originHostName
+
+    $endpoint = Get-AzCdnEndpoint -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName
+    $validateResult = Test-AzCdnCustomDomain -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName -CustomDomainHostName $hostName
+    Assert-True{$validateResult.CustomDomainValidated}
+    $validateResultbyPiping = Test-AzCdnCustomDomain -CdnEndpoint $endpoint -CustomDomainHostName $hostName
+    Assert-True{$validateResultbyPiping.CustomDomainValidated}
+
+    $createdCustomDomain = New-AzCdnCustomDomain -HostName $hostName -CustomDomainName $customDomainName -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName
+    Assert-AreEqual $customDomainName $createdCustomDomain.Name
+    Assert-AreEqual $hostName $createdCustomDomain.HostName
+
+    $customDomain = Get-AzCdnCustomDomain -CustomDomainName $customDomainName -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName
+    Assert-AreEqual $customDomainName $customDomain.Name
+    Assert-AreEqual $hostName $customDomain.HostName
+
+    $enabled = $customDomain | Enable-AzCdnCustomDomainHttps -PassThru
+    Assert-True{$enabled}
+
+    $customDomain = Get-AzCdnCustomDomain -CustomDomainName $customDomainName -EndpointName $endpointName -ProfileName $profileName -ResourceGroupName $resourceGroup.ResourceGroupName
+
+    Assert-AreEqual $customDomain.CustomHttpsProvisioningState "Enabling"
```

Comment:
Can you add a positive test for Disable-AzCdnCustomDomainHttps as well?

Reply:
Its a bit tricky to add a positive test for Disable-AzCdnCustomDomainHttps as calling Disable requires a custom domain https to be enabled first. But it takes a few hours for the enable Custom Domain Https to complete.

Reply:
That is fine - only the recording will take a few hours, and that occurs very rarely (when the API is bumped).  During playback it should only take a couple of seconds. This way you can also test that the enable functionality completes correctly with the cmdlet (not just that it starts).

Reply:
Added the test.